src/dataMigration.js

```diff
@@ -52,4 +54,18 @@ const dataMigrations = [
       if (syncURL && syncURL.length > 0) settings.set(SETTINGS_KEYS.SYNC_IS_INITIALISED, 'true');
     },
   },
+  {
+    version: '2.0.0',
+    migrate: database => {
+      // Changed SyncQueue to expect no more than one SyncOut record for every record in database.
+      // Assume that last SyncOut record is correct.
+      const allRecords = database.objects('SyncOut').sorted('changeTime');
+      database.write(() => {
+        allRecords.forEach(record => {
+          const hasDuplicates = allRecords.filtered('recordId == $0', record.id).length > 0;
+          if (hasDuplicates) database.delete('SyncOut', record);
+        });
+      });
+    },
+  },
```

Comment:
Well it'll currently just deletes the whole queue because `> 0` haha. Given changing to `> 1`. I got what you mean. It's a bit dirty but I could just add all record type 'Transaction' (and requisitions?) SyncOuts to the end of the queue right?